### DISCUSSION TITLE
Provide an API endpoint to enable ASN object assignments

### DISCUSSION THREAD

**Discussion by richardjohnallsopp** [netbox-community/netbox#14122] [Help Wanted!]
### NetBox version

v3.6.2

### Feature type

Change to existing functionality

### Proposed functionality

It could be something I'm missing, but I don't see a way to assign an ASN object to a site like you can with contact objects.

### Use case

We're in the process of migrating from v2.11.11 to v3.6.2 and there are several changes to the way data is handled. We're writing some scripts to handle these changes like pulling contacts from our current version and creating + assigning them on our updated stage version. We don't see a way to handle ASN objects programatically using the API and Pynetbox.

For example with contacts we were able to do this:
```
import pynetbox

nb_prod = pynetbox.api(url='https://netbox.prod, token='')
nb_dev = pynetbox.api(url='https://netbox.stage', token='')

# fetch all site objects
nb_site_list = nb_prod.dcim.sites.all()

def assigning_site_contacts(data,api):

  for site in data:
    if site.contact_name != "":
      assigning_contacts = api.tenancy.contact_assignments.create(
          content_type = "dcim.site",
          object_id = site.id,
          contact = {'name': site.contact_name},
          role = {'name': 'Site Contact'},
      )
    print("assigning contacts: " + str(assigning_contacts))

assigning_site_contacts(nb_site_list,nb_dev)

```  
Could we add something like this for ASN objects?

**Comment by DanSheps** [comment]
Site has a field directly for asn assignment, it is "asns".

The reason contacts have a separate API endpoint is because contacts are a complex field (Many-to-many using a generic foreign key -- a generic foreign key is a link to **any** table in the database) whereas ASN uses a standard M2M relationship.
